cmd/cloudx/client/handler_test.go

```diff
@@ -58,6 +58,222 @@ func TestCommandHelper(t *testing.T) {
+
+		t.Run("With some projects returns list of projects", func(t *testing.T) {
+			cmd := *cmd_base
+			project_name1 := "new_project_name1"
+			project_name2 := "new_project_name2"
+
+			project1, err := cmd.CreateProject(project_name1, false)
+			require.NoError(t, err)
+			project2, err := cmd.CreateProject(project_name2, false)
+			require.NoError(t, err)
+
+			projects, err := cmd.ListProjects()
+
+			require.NoError(t, err)
+			assert.Len(t, projects, 2)
+			listedIds := [2]string{projects[0].Id, projects[1].Id}
+			assert.Contains(t, listedIds, project1.Id)
+			assert.Contains(t, listedIds, project2.Id)
+		})
+	})
+
+	t.Run("func=CreateProject", func(t *testing.T) {
+		configDir := testhelpers.NewConfigDir(t)
+		testhelpers.RegisterAccount(t, configDir)
+
+		cmd_base := &client.CommandHelper{
+			ConfigLocation:   configDir,
+			NoConfirm:        true,
+			IsQuiet:          false,
+			VerboseWriter:    io.Discard,
+			VerboseErrWriter: io.Discard,
+			Ctx:              context.Background(),
+		}
+
+		t.Run("creates project and sets default project", func(t *testing.T) {
+			cmd := *cmd_base
+			project_name := "new_project_name"
+
+			project, err := cmd.CreateProject(project_name, true)
+			require.NoError(t, err)
+			assert.Equal(t, project.Name, project_name)
+
+			defaultId, err := cmd.GetDefaultProjectID()
+			require.NoError(t, err)
+			assert.Equal(t, project.Id, defaultId)
+		})
+
+		t.Run("creates two projects with different names", func(t *testing.T) {
+			cmd := *cmd_base
+			project_name1 := "new_project_name1"
+			project_name2 := "new_project_name2"
+
+			project1, err := cmd.CreateProject(project_name1, true)
+			require.NoError(t, err)
+
+			project2, err := cmd.CreateProject(project_name2, false)
+			require.NoError(t, err)
+
+			assert.NotEqual(t, project1.Id, project2.Id)
+			assert.NotEqual(t, project1.Name, project2.Name)
+			assert.NotEqual(t, project1.Slug, project2.Slug)
+
+			defaultId, err := cmd.GetDefaultProjectID()
+			require.NoError(t, err)
+			assert.Equal(t, project1.Id, defaultId)
+		})
+	})
+
+	t.Run("func=Authenticate", func(t *testing.T) {
+		cmd_base := &client.CommandHelper{
+			ConfigLocation:   testhelpers.NewConfigDir(t),
+			NoConfirm:        true,
+			IsQuiet:          false,
+			VerboseWriter:    io.Discard,
+			VerboseErrWriter: io.Discard,
+			Ctx:              context.Background(),
+		}
+
+		t.Run("create new account", func(t *testing.T) {
+			cmd := *cmd_base
+
+			name := testhelpers.FakeName()
+			email := testhelpers.FakeEmail()
+			var r bytes.Buffer
+			_, _ = r.WriteString("n\n")        // Do you want to sign in to an existing Ory Network account? [y/n]: n
+			_, _ = r.WriteString(email + "\n") // Email: FakeEmail()
+			_, _ = r.WriteString(name + "\n")  // Name: FakeName()
+			_, _ = r.WriteString("n\n")        // Subscribe to the Ory Security Newsletter to get platform and security updates? [y/n]: n
+			_, _ = r.WriteString("y\n")        // I accept the Terms of Service [y/n]: y
```

Comment:
This is quite similar to `testhelpers.RegisterAccount()`. Can't you just use that function, or else split the functionality in that package? I feel like we only want to maintain this interaction once.

Reply:
The similarity stems primarily from the definition of the input buffer. Here I'm actually testing the logic of `Authenticate()` without going through `cobra.Command` like the `testhelpers.RegisterAccount()` does.
One could maybe move the logic into a separate function, but in my opinion this would make the tests more difficult to understand, so I would leave this as is.
WDYT?